app/uk/gov/hmrc/apinotificationqueue/controller/EnhancedNotificationsController.scala

```diff
@@ -49,10 +49,32 @@ class EnhancedNotificationsController @Inject()(queueService: QueueService,
   def getPulledByClientId: Action[AnyContent] = pullByClientId(Pulled)
   def getUnpulledByClientId: Action[AnyContent] = pullByClientId(Unpulled)
 
-  private def pullByClientId(notificationStatus: NotificationStatus.Value): Action[AnyContent] = Action.async { implicit request =>
+  def getPulledByConversationId(conversationId: UUID): Action[AnyContent] = pullByConversationId(conversationId: UUID, Pulled)
+  def getUnpulledByConversationId(conversationId: UUID): Action[AnyContent] = pullByConversationId(conversationId: UUID, Unpulled)
+
+  def getByConversationId(conversationId: UUID): Action[AnyContent] = Action.async { implicit request =>
 
     val headers = request.headers
-    logger.info(s"listing $notificationStatus notifications", headers.headers)
+    logger.info(s"listing notifications with conversationId $conversationId", headers.headers)
+    validateClientIdHeader(request.headers, "getByConversationId") match {
+      case Left(errorResponse) => Future.successful(errorResponse.XmlResult)
+      case Right(clientId) =>
+        val notificationIdPaths: Future[List[String]] =
+          for {
+            notificationIds <- queueService.getByConversationId(clientId, conversationId)
+          } yield notificationIds.map { n =>
+            val status = if (n.pulledStatus) "pulled" else "unpulled"
```

Comment:
This would also read better as `if (n.pulled) "pulled" else "unpulled"`

Reply:
Done